src/wp-includes/class-wp-theme.php

```diff
@@ -1993,16 +1997,43 @@ private function set_pattern_cache( array $patterns ) {
 			'version'  => $this->get( 'Version' ),
 			'patterns' => $patterns,
 		);
-		wp_cache_set( 'wp_theme_patterns_' . $this->stylesheet, $pattern_data, 'theme_files' );
+
+		/**
+		 * Filters the cache expiration time for theme files.
+		 *
+		 * @since 6.6.0
+		 *
+		 * @param int    $cache_expiration Cache expiration time in seconds.
+		 * @param string $cache_type       Type of cache being set.
+		 */
+		$cache_expiration = apply_filters( 'wp_theme_files_cache_ttl', self::$cache_expiration, 'theme_block_patterns' );
+
+		// We don't want to cache patterns infinitely.
+		if ( ! is_int( $cache_expiration ) || $cache_expiration <= 0 ) {
```

Comment:
The `is_int()` check is unnecessarily strict, it doesn't bring any real benefit. I think for that we need to be more forgiving and simply cast to an integer.
```suggestion
		$cache_expiration = (int) apply_filters( 'wp_theme_files_cache_ttl', self::$cache_expiration, 'theme_block_patterns' );

		// We don't want to cache patterns infinitely.
		if ( $cache_expiration <= 0 ) {
```